src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.DecisionDagRewriter.cs

```diff
@@ -138,7 +138,9 @@ public override BoundNode VisitCall(BoundCall node)
                         // might be a call to a local function that assigns something
                         node.Method.MethodKind == MethodKind.LocalFunction ||
                         // or perhaps we are passing a variable by ref and mutating it that way
-                        !node.ArgumentRefKindsOpt.IsDefault;
+                        !node.ArgumentRefKindsOpt.IsDefault ||
+                        // or perhaps we are calling a mutating method of a value type
+                        !node.Method.IsStatic && node.Method.ContainingType.IsStructType() && node.Method.ContainingType.SpecialType == SpecialType.None;
```

Comment:
Should we refrain from setting this flag when the method is `readonly`? e.g. `!node.Method.IsStatic && !node.Method.IsEffectivelyReadOnly && ...`

Reply:
It isn't just about mutating the receiver.  A method on the type `Span<T>` can mutate some other variable.  Though frankly, so can a method on a class type (by invoking a delegate).